articles/vue2-crud-app-with-localstorage/index.md

```diff
@@ -0,0 +1,515 @@
+```javascript
+name: "AddBookItem",
+props: ['editBook'],
+data () {
+    return {
+        title: '',
+        id: '',
+        edit: false
+    }
+}
+```
+We will use this `edit` property to decide whether to edit or add a new book. We first check if the user is not editing. We save the data, else, we edit the data. If we’re editing, we emit an `edit-book-event` and pass the variable that holds the edited data `bookItem` along with the event to the parent. We also clear the input field. Now update `addBookItem()` method to look as shown below.
+```javascript
+addBook(e){
+    e.preventDefault();
+    if (this.edit === false){
+        // add new book
+        const newBook = {
+            title: this.title,
+            id: Math.floor(Math.random() * 100)
+        };
+        if (newBook.title !== ''){
+            this.$emit('add-book-event', newBook);
+        }
+        this.title = ''
+    }else{
+        //edit book
+        const bookItem = {
+            title: this.title,
+            id: this.id
+        }
+        //send to parent (App.vue)
+        this.$emit('edit-book-event', bookItem)
+        // clear input field
+        this.title = '';
+        this.edit = false;
+    }
+}
+``` 
+Now you can click on the edit button and the input field will be populated with the book title. The `watch()` method comes in handy again, to help us watch for any changes in the editBook data. We set `deep:true` property to let the Vue instance continuously watch for changes. So, while editing a book, the `edit` property will always be true.
+
+It also watches the title property and if it’s empty, it sets the `edit` property to false. Here we don’t need the `deep` property.
+```javascript
+watch: {
+    editBook: {
+        handler() {
+            this.title = this.editBook.title;
+            this.id = this.editBook.id;
+            this.edit = true
+        },
+        deep: true
+    },
+    title: {
+        handler() {
+            if (this.title === ''){
+                this.edit = false;
+            }
+        }
+    }
+}
+```
+Back to `App.vue`, after editing a title, an event `edit-book-event` is sent to `App.vue`. We assign the event to a method in order to save the changes to local storage. Update your code to look like shown below.
+```html
+<AddBookItem v-model="editBook.title" v-on:add-book-event="addBookItem"  v-bind:editBook="editBook" v-on:edit-book-event="editBookItemEvent" />
+```
+Now we create a method `editBookItemEvent()` to handle saving of data. In the method we find the index of the ID’s object. This index will be used to reassign the title of the book being edited. If you’ve reached this far you can edit a book title.  Add the code below to `methods` in `App.vue`.
+```javascript
+editTodoItemEvent(todoItem){
+    //find index of this id's object
+    var objIndex = this.todo_items.findIndex(obj => obj.id === todoItem.id);
+    //update the item
+    this.todo_items[objIndex].title = todoItem.title;
+}
```

Comment:
Looks like you are using the wrong block of code here. Please replace this with the correct books code.